src/node/desktop/src/main/args-manager.ts

```diff
@@ -146,8 +146,12 @@ export class ArgsManager {
             setenv(kRStudioInitialProject, arg);
             return false;
           }
+          else {
```

Comment:
for all files that are not Rproj files

Reply:
You might be able to do the same thing for the `open-file` event. However, the event may be called at any time and not necessarily before the session launches. The electron docs specifies that the `open-file` event at startup needs to be handled before the app is ready. But I'm not sure if that guarantees that opening from Finder will always fire the event before app ready.

I guess try that in the next PR and find out! 🤞

Reply:
yep, working on that PR now -- seems like we can just have an if/else clause based on whether or not the app is ready yet. That's what we do for handling the Rproj files so I am guessing the same thing will work for other files